content/posts/2025-02-26-whats_new_1_8.md

```diff
@@ -0,0 +1,295 @@
+      - UDNMapping
+```
+
+<p style="text-align: center">Listing 8: Enable UDNMapping feature in FlowCollector</p>
+
+Now let's create a user-defined network.  You can use `oc apply` with the content in Listing 9, or copy and paste this into OpenShift web console.  To do the latter, click the **+** icon in the upper right corner next to the **?** icon, and select **Import YAML**.  Paste the YAML in and click **Create**.  You will now see the entry in **Networking > UserDefinedNetworks**.  Note: While you can create the UDN from this panel, it is very limiting in what you can configure.
+
+```
+apiVersion: k8s.ovn.org/v1
+kind: UserDefinedNetwork
+metadata:
+  name: udn-chiefs
+  namespace: chiefs
+spec:
+  topology: Layer2
+  layer2:
+    role: Primary
+    subnets:
+      - "10.0.0.0/24"
+```
+
+<p style="text-align: center">Listing 9: UDN - Layer 2 network segment</p>
+
+The next step is to add a secondary interface to a pod and have it join the UDN in the secondary network.  This is done using an annotation.  Set **k8s.v1.cni.cncf.io/networks** to the name of the UDN.  Assuming the pod is in a deployment, delete the pod and it will be recreated with two interfaces.  Check that there is a secondary interface.  See the commands in Listing 10.
+
+```
+pod="your_pod_name"  # replace with your pod name
+oc annotate pod/$pod k8s.v1.cni.cncf.io/networks=udn-chiefs
+> Delete pod.  It will be recreated with two interfaces.
+oc describe pod/$pod  # should see two interfaces mentioned in Annotations
+```
+
+<p style="text-align: center">Listing 10: Add secondary interface to pod</p>
+
+Once it joins the UDN, it becomes isolated from all other pods except those that join the secondary network.  From the UDN configuration, it gets an IP address  from 10.0.0.0/24.  In essence, a namespace could be a tenant.
+
+In Network Observability, the Traffic flow table has a UDN labels column (Figure 4).  You can filter on Source Network Name and Destination Network Name.
+
+![Flows table with UDN labels]({page.image('whats-new-1-8/flows-udn.png')})
+Figure 4: Flows table with UDN labels
+
+In the Topology tab, there is a new top-level scope called "Net" (Network).  This shows all of your secondary networks.  Figure 5 shows two UDNs.
+
+![Topology showing UDNs]({page.image('whats-new-1-8/topo-udn.png')})
+Figure 5: Topology showing UDNs
+
+
+### eBPF Manager Support
+
+With the proliferation of eBPF programs in networking, monitoring, tracing, and security, there is potential for conflicts in the use of the same eBPF hooks.  The eBPF Manager is a separate operator that manages all eBPF programs and ensures compliance, security, and prevents conflicts.  Network Observability can leverage eBPF Manager and let it handle the loading of hooks, ultimately removing the need to provide the eBPF agent with *privileged* mode.  Since this is a Developer Preview feature, *privileged* mode is still required for now and it is only supported on the amd64 architecture.
```

Comment:
not only the privileged mode, but also all its linux capabilities (CAP_BPF etc.)
So even when it's not run as privileged, ultimately it will be able to run just as a standard pod.

That's really great for reducing attack surface especially when there's more than 1 ebpf-based product installed.